File: open-agent-capsule-starter-complete/examples/hooks/go/post_hydrate_notice.go
package main

import (
    "encoding/json"
    "os"
)

type payload struct {
    Event  string `json:"event"`
    Tool   string `json:"tool"`
    Status string `json:"status"`
    Note   string `json:"note"`
}

func main() {
    out := payload{
        Event:  "post-hydrate",
        Tool:   "oac",
        Status: "ok",
        Note:   "go starter hook placeholder",
    }
    _ = json.NewEncoder(os.Stdout).Encode(out)
}

File: open-agent-capsule-starter-complete/examples/hooks/rust/post_hydrate_notice.rs
//! Small example hook used by starter adapter profiles.
//!
//! This starter keeps the Rust example dependency-free and intentionally boring.
//! In a real setup you would parse the hook payload, validate sidecars, or notify an
//! external system after an opt-in hydrate step.

use std::io::{self, Write};

fn main() {
    let payload = r#"{"event":"post-hydrate","tool":"oac","status":"ok","note":"rust starter hook placeholder"}"#;
    let mut stdout = io::stdout();
    stdout.write_all(payload.as_bytes()).expect("write stdout");
    stdout.write_all(b"
").expect("write newline");
}
